judge/models/profile.py

```diff
@@ -121,6 +123,10 @@ class Profile(models.Model):
                                       help_text=_('32 character base32-encoded key for TOTP'),
                                       validators=[RegexValidator('^$|^[A-Z2-7]{32}$',
                                                                  _('TOTP key must be empty or base32'))])
+    scratch_codes = EncryptedNullCharField(max_length=255, null=True, blank=True, verbose_name=_('scratch codes'),
+                                           help_text=_('JSON of 16 character base32-encoded codes for scratch codes'),
+                                           validators=[RegexValidator('^[A-Z2-7]$',
```

Comment:
Is this regex correct? Doesn't the JSON also contain `[`, `]`, and `,`, for example? I think right now it works because calling `save` directly bypasses the validation, but you won't be able to save on the admin site.

On a related note, should this field be added to the admin site with similar permissions required as `totp_key`?